pkg/auth/manager.go

```diff
@@ -0,0 +1,315 @@
+	sort.Strings(clusterNames)
+
+	// Build htpasswd content from entries
+	var htpasswdLines []string
+	for _, clusterName := range clusterNames {
+		if clusterAuthBytes, exists := authSecret.Data[clusterName]; exists {
+			var clusterAuthData ClusterAuthData
+			if err := json.Unmarshal(clusterAuthBytes, &clusterAuthData); err != nil {
+				return "", fmt.Errorf("failed to unmarshal cluster data for %s: %w", clusterName, err)
+			}
+
+			// Use cached htpasswd from JSON structure
+			if clusterAuthData.Htpasswd == "" {
+				return "", fmt.Errorf("missing htpasswd entry for cluster %s", clusterName)
+			}
+			htpasswdLines = append(htpasswdLines, clusterAuthData.Htpasswd)
+		}
+	}
+
+	return strings.Join(htpasswdLines, "\n"), nil
+}
+
+// createOrUpdateHtpasswdSecret creates or updates an htpasswd secret with the given name and data key
+func (am *authManager) createOrUpdateHtpasswdSecret(ctx context.Context, secretName, dataKey string) error {
+	logger := log.FromContext(ctx)
+	htpasswdSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: am.config.SecretsNamespace,
+		},
+	}
+
+	result, err := ctrl.CreateOrUpdate(ctx, am.client, htpasswdSecret, func() error {
+		// Generate htpasswd content
+		htpasswdContent, err := am.generateHtpasswdContent(ctx)
+		if err != nil {
+			return err
+		}
+
+		// Initialize or update the secret data
+		if htpasswdSecret.Data == nil {
+			htpasswdSecret.Data = make(map[string][]byte)
+		}
+		htpasswdSecret.Data[dataKey] = []byte(htpasswdContent)
+
+		return nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to create or update %s secret: %w", secretName, err)
+	}
+
+	logger.Info("Htpasswd secret processed", "secret", secretName, "result", result)
+	return nil
+}
+
+// AddClusterPassword adds a password for a specific cluster to the auth secret
+func (am *authManager) AddClusterPassword(ctx context.Context, clusterName string) error {
+	logger := log.FromContext(ctx)
+	authSecret := am.getAuthSecret()
+
+	result, err := ctrl.CreateOrUpdate(ctx, am.client, authSecret, func() error {
+		// Initialize Data map if it doesn't exist
+		if authSecret.Data == nil {
+			authSecret.Data = make(map[string][]byte)
+		}
+
+		// Check if cluster already has a password
+		if _, exists := authSecret.Data[clusterName]; exists {
+			return nil // Already exists, no changes needed
+		}
+
+		// Generate password for new cluster
+		password, err := am.passwordGenerator.GeneratePassword(32)
+		if err != nil {
+			return fmt.Errorf("failed to generate password for cluster %s: %w", clusterName, err)
+		}
+
+		// Generate htpasswd entry
+		htpasswdEntry, err := am.passwordGenerator.GenerateHtpasswd(clusterName, password)
+		if err != nil {
+			return fmt.Errorf("failed to generate htpasswd for cluster %s: %w", clusterName, err)
+		}
+
+		// Store both password and htpasswd in JSON format
+		clusterData := ClusterAuthData{
+			Password: password,
+			Htpasswd: htpasswdEntry,
+		}
```

Comment:
Storing the encrypted and non-encrypted password in the ClusterAuthData struct feels wrong somehow.